diff -ur indicator-applet-0.4.12/configure.ac indicator-applet-0.4.12.new/configure.ac
--- indicator-applet-0.4.12/configure.ac	2011-04-15 06:32:21.000000000 +0400
+++ indicator-applet-0.4.12.new/configure.ac	2011-08-15 22:07:11.764747702 +0400
@@ -31,7 +31,7 @@
 PKG_CHECK_MODULES(APPLET, gtk+-2.0 >= $GTK_REQUIRED_VERSION
                           x11
                           libpanelapplet-3.0
-                          indicator >= $INDICATOR_REQUIRED_VERSION)
+                          indicator-0.4 >= $INDICATOR_REQUIRED_VERSION)
 AC_SUBST(APPLET_CFLAGS)
 AC_SUBST(APPLET_LIBS)
 
